dpnp/dpnp_algo/dpnp_algo_trigonometric.pyx

```diff
@@ -116,8 +162,31 @@ cpdef dparray dpnp_expm1(utils.dpnp_descriptor x1):
     return call_fptr_1in_1out(DPNP_FN_EXPM1, x1, x1.shape)
 
 
-cpdef dparray dpnp_log(utils.dpnp_descriptor x1):
-    return call_fptr_1in_1out(DPNP_FN_LOG, x1, x1.shape)
+cpdef dparray dpnp_log(utils.dpnp_descriptor x1, dparray out=None):
```

Comment:
I think it is better to change `call_fptr_1in_1out` to add one more parameter `out`.